internal/helm/chart.go

```diff
@@ -0,0 +1,89 @@
+package helm
+
+import (
+	"fmt"
+
+	"github.com/airbytehq/abctl/internal/common"
+	"github.com/airbytehq/abctl/internal/validate"
+	goHelm "github.com/mittwald/go-helm-client"
+	"golang.org/x/mod/semver"
+)
+
+// ChartResolver handles resolution of Airbyte chart references from v1/v2 repositories, local paths, and URLs
+type ChartResolver struct {
+	// v1RepoURL is the base URL for v1 Airbyte helm charts
+	v1RepoURL string
+	// v2RepoURL is the base URL for v2 Airbyte helm charts
+	v2RepoURL string
+	// client provides Chart.yaml metadata extraction for local paths and URLs
+	client goHelm.Client
+}
+
+// NewChartResolver creates a resolver with default Airbyte v1/v2 repository URLs
+func NewChartResolver(client goHelm.Client) *ChartResolver {
+	return &ChartResolver{
+		v1RepoURL: common.AirbyteRepoURLv1,
+		v2RepoURL: common.AirbyteRepoURLv2,
+		client:    client,
+	}
+}
+
+// NewChartResolverWithURLs creates a resolver with custom v1/v2 repository URLs
+func NewChartResolverWithURLs(client goHelm.Client, v1URL, v2URL string) *ChartResolver {
+	return &ChartResolver{
+		v1RepoURL: v1URL,
+		v2RepoURL: v2URL,
+		client:    client,
+	}
+}
+
+// ChartIsV2Plus returns true if the chart version is v2.0.0 or higher
+func ChartIsV2Plus(v string) bool {
+	if v == "" {
+		return false
+	}
+	if v[0] != 'v' {
+		v = "v" + v
+	}
+	return semver.Compare(v, "v2.0.0") >= 0
+}
+
+// ResolveChartReference resolves an Airbyte chart reference to its full URL/path and version.
+// For empty chart+version, returns latest v2 chart.
+// For version-only, uses v1/v2 repo based on version number.
+// For URLs and local paths, returns as-is with version from chart metadata.
+func (r *ChartResolver) ResolveChartReference(chart, version string) (string, string, error) {
+	if chart == "" {
```

Comment:
Could negate this, return early, and remove a layer of indentation
```
if chart != "" { 
  return
}
```

Reply:
Re-reading this function.  What am I missing?  It returns a `(string, string, error)` but it doesn't look like all the branches return... i.e. if `chart != ""`, what happens?

Reply:
I'll update the doc, they are not clear. First is the chart path/location, second is the chart version.